src/utils.js

```diff
@@ -391,6 +400,56 @@ async function validateAndInterpolateMesh(inputMeshData, envFilePath, command) {
 	}
 }
 
+/**
+ * Validate secrets file
+ *
+ * @param secretsFile Validates that secrets file extension is in yaml
+ */
+function validateSecrets(secretsFile) {
+	try {
+		const validExtensions = ['.yaml', '.yml'];
+		const fileExtension = secretsFile.split('.').pop().toLowerCase();
+		if (!validExtensions.includes('.' + fileExtension)) {
+			throw new Error('Invalid file format. Please provide a YAML file (.yaml or .yml).');
+		}
+	} catch (error) {
+		logger.error(error.message);
+		throw new Error(error.message);
+	}
+}
+
+/**
+ * Read the secrets file, checks validation and interpolate mesh
+ *
+ * @param secretsFilePath Secrets file path
+ * @param command
+ */
+async function interpolateSecrets(secretsFilePath, command) {
+	try {
+		validateSecrets(secretsFilePath);
```

Comment:
Please move the call to the validate function outside of the interpolateSecrets function. Call the validate function from create / update functions. 

Its always better to make utility functions atomic (no side effects).

Reply:
Done, moved validate function outside.